packages/@cdktf/hcl2cdk/lib/expressions.ts

```diff
@@ -12,101 +13,54 @@ export type Reference = {
   isVariable?: boolean;
 };
 
-const PROPERTY_ACCESS_REGEX = /\[.*\]/;
 const DOLLAR_REGEX = /\$/g;
 
-export function extractReferencesFromExpression(
+export async function extractReferencesFromExpression(
   input: string,
   nodeIds: readonly string[],
   scopedIds: readonly string[] = [] // dynamics introduce new scoped variables that are not the globally accessible ids
-): Reference[] {
-  const isDoubleParanthesis = input.startsWith("${{");
-  if (!input.startsWith("${")) {
-    return [];
-  }
-
-  const start = isDoubleParanthesis ? 3 : 2;
-  const end = isDoubleParanthesis ? input.length - 2 : input.length - 1;
-  let expressionString = input.substring(start, end);
+): Promise<Reference[]> {
+  logger.debug(`extractReferencesFromExpression(${input})`);
+  const possibleVariableSpots = await getReferencesInExpression(
+    "main.tf",
+    input
+  );
 
-  if (
-    expressionString.includes("for") &&
-    expressionString.includes("in") &&
-    expressionString.includes(":")
-  ) {
-    // for name, user in var.users : user.role => name...
-    // We just want the var.users part (that could be an expression)
-    expressionString = expressionString.substring(
-      expressionString.indexOf("in") + 2,
-      expressionString.indexOf(":")
-    );
-  }
-  const lines = expressionString
-    .split("\n")
-    .map((line) => {
-      const commentStart = line.indexOf("#");
-      const lineWithoutComment =
-        commentStart !== -1 ? line.substring(0, commentStart - 1) : line;
-
-      return lineWithoutComment.trim();
-    })
-    .filter((line) => line !== "");
-
-  const delimiters = [
-    "(",
-    ",",
-    ")",
-    ".*",
-    PROPERTY_ACCESS_REGEX,
-    " ",
-    "!",
-    "*",
-    "/",
-    "%",
-    ">",
-    "<",
-    "=",
-    "&&",
-    "||",
-    "?",
-    // There can be nested terraform expression strings
-    "${",
-    "}",
-  ];
-
-  let possibleVariableSpots = lines;
-
-  delimiters.forEach((delimiter) => {
-    possibleVariableSpots = possibleVariableSpots.reduce(
-      (carry, str) => [...carry, ...str.split(delimiter)],
-      [] as string[]
-    );
-  });
+  logger.debug(
+    `found possible variable spots: ${JSON.stringify(possibleVariableSpots)}`
+  );
```

Comment:
I love how much code is deleted here 😊

Reply:
yeah, it was the most fragile code as well :)